metaflow/plugins/kfp/kfp.py

```diff
@@ -586,17 +575,9 @@ def _set_container_resources(
             container_op.container.set_ephemeral_storage_limit(
                 resource_requirements["local_storage_limit"]
             )
-        if (
-            kfp_component.pytorch_distributed_decorator
-            or "volume" in resource_requirements
-        ):
-            if kfp_component.pytorch_distributed_decorator:
-                print("This is now deprecated!")
-                mode = [VOLUME_MODE_RWM]
-                volume_dir = "/opt/pytorch_shared/"
-            else:
-                mode = resource_requirements["volume_mode"]
-                volume_dir = resource_requirements["volume_dir"]
+        if "volume" in resource_requirements:
+            mode = resource_requirements["volume_mode"]
```

Comment:
KFP doesn't validate, and neither does this resource decorator validate for it's existing values such as `cpu`.  This could be because it is meta and each system is different.